File: AuraFit-BackEnd/src/main/java/app/aurafitbackend/Beans/Promotion.java
package app.aurafitbackend.Beans;

import jakarta.persistence.*;
import lombok.*;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.time.LocalDateTime;

@Entity
@Table(name = "promotions")
@NoArgsConstructor
@AllArgsConstructor
@Data
@Builder
public class Promotion {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Setter(AccessLevel.NONE)
    private Long id;
    private String name;
    private Integer discountPercent;
    private LocalDateTime startTime;
    private LocalDateTime endTime;
    private Boolean isActive = false;
    @ManyToOne
    private ProductVariant productVariant;


}

File: aurafit-frontend/src/Models/Promotion.ts
import {Product} from "./Product.ts";
import {ProductVariant} from "./ProductVariant.ts";

export class Promotion {
    id: number
    name: string
    discountPercent: number
    startTime: Date
    endTime: Date
    isActive: boolean
    productVariant: ProductVariant

    constructor(id: number, name: string, discountPercent: number, startTime: Date, endTime: Date, isActive: boolean, productVariant: ProductVariant) {
        this.id = id
        this.name = name
        this.discountPercent = discountPercent
        this.startTime = startTime
        this.endTime = endTime
        this.isActive = isActive
        this.productVariant = productVariant
    }
}